privacy_consent/views/res_partner.xml

```diff
@@ -6,16 +6,13 @@
         <field name="name">Add consent smart button</field>
         <field name="model">res.partner</field>
         <field name="inherit_id" ref="base.view_partner_form" />
-        <field
-            name="groups_id"
-            eval="[(4, ref('privacy.group_data_protection_user'))]"
-        />
         <field name="arch" type="xml">
             <div name="button_box" position="inside">
                 <button
                     attrs='{"invisible": [("privacy_consent_count", "=", 0)]}'
                     class="oe_stat_button"
                     context='{"search_default_partner_id": active_id}'
+                    groups="privacy.group_data_protection_user"
```

Comment:
Groups cannot be applied on inherited views since odoo v16 (only on base views), allowing to simplify inheritance computation (all inherited views always apply, maybe with a void diff but always present). That's why I moved it on the field definition itself.

Reply:
But then, how can you overcome the permission problems?

Reply:
With groups on the field it will not try to compute it. The inherit is present in the hierarchy but not the field itself. Globally idea is to move groups on view inherits to part of the arch that we want to protect (even if it consists in the whole arch of the inherit).

Reply:
For more details, you can check work of Denis at https://github.com/odoo/odoo/pull/98551 and https://github.com/odoo/odoo/pull/95729 notably :)

Reply:
@yajo can you please re-check this?